TokenProvider/ContentDateTimeProvider.php

```diff
@@ -47,16 +49,25 @@ public function configureOptions(OptionsResolverInterface $optionsResolver)
             'date_format' => 'Y-m-d',
         ));
 
-        $optionsResolver->setAllowedTypes(array(
-            'date_format' => 'string',
-        ));
-        
-        // The slugify option is deprecated as of version 1.1 and will be removed in 2.0. Setting it doesn't change anything
-        if (!method_exists($optionsResolver, 'setDefined')) {
-            // To support Symfony <2.6
-            $optionsResolver->setOptional(array('slugify'));
+        if ($newApi) {
+            $optionsResolver->setAllowedTypes('date_format', 'string');
         } else {
+            $optionsResolver->setAllowedTypes(array(
+                'date_format' => 'string',
+            ));
+        }
+
+        // The slugify option is deprecated as of version 1.1 and will be removed in 2.0. Setting it doesn't change anything
+        if ($newApi) {
             $optionsResolver->setDefined(array('slugify'));
```

Comment:
Mostly because I (a) found the variable more self-explaining and (b) did not want to copy the deprecation comment.

However, updated it to one if statement